### PR TITLE
🩹 Raise a custom exception instead of `RuntimeError`

### LINKED ISSUE #3540: Replace raw RuntimeError with semantic exception when a service is not configured
From the PR review discussion in #3490

Places where raw `RuntimeError` is used rather than a custom `NoServiceConfigured` exception class:

* `openforms.appointments.contrib.qmatic.client.QmaticClient`

Seems to be the only place. Other raw `RuntimeError` is semantically correct.

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #3540.

### CODE REVIEW

File: src/openforms/appointments/contrib/qmatic/client.py [hunk 1]
```diff
@@ -73,7 +77,7 @@ def QmaticClient() -> "Client":
     config = QmaticConfig.get_solo()
     assert isinstance(config, QmaticConfig)
```
Comment (Viicos) [inline src/openforms/appointments/contrib/qmatic/client.py:78]: Should `cast` be used instead? Adds an extra import though, I guess this is just a personal preference

Comment (sergei-maertens) [reply src/openforms/appointments/contrib/qmatic/client.py:78]: no, @CharString has strong feelings about that :)

**Review by sergei-maertens — APPROVED** [review]

File: src/openforms/appointments/contrib/qmatic/client.py (hunk 1 above)
Comment (Viicos) [reply src/openforms/appointments/contrib/qmatic/client.py:78]: I'm curious to know why :)

Comment (sergei-maertens) [reply src/openforms/appointments/contrib/qmatic/client.py:78]: `cast` is a lie and the runtime behaviour can differ from the declared type, while `assert` enforces runtime correctness too. If assert is causing too much overhead, we can also strip it out and run with `python -O`

basically `cast` is a last resort if there are really no better options (and I agree)

**Pull request merged**